### DISCUSSION TITLE
add tivo file id as one of the variables in the file naming drop down

### DISCUSSION THREAD

**Discussion by tweiner** [lart2150/kmttg#27] [Ideas]
I have some downloads that the file naming option make them identical.  Would it be possible to add some of the tivo internal flags like the tivo file id.  It is a unique number that would be helpful.

**Comment by lart2150** [comment]
I like this idea.  I could add something like [recordingId] you could also likely use [startTime] unless you have two recordings that started at the same time.

**Reply by tweiner** [reply]
Thank you.  Recording ID would be perfect.  I may be getting ahead of myself how difficult would it be to append a copy # at the end of a duplicate file name like (1) or _1.  That could be an option for "do not overwrite".  I think I am getting a bit greedy how about [duration].

Thank you for your response.

**Reply by lart2150** [reply]
adding a toggle to no overwrite would not be a huge change but it would be a bigger change then I feel like coding right now 🙂.  I just pushed a change that adds support for [recordingId] [contentId] [collectionId].  recording id should be unique for a given recording.  content id I think should be unique to a episode and collectionId should be unique to a tv show or movie.

[kmttg_20240720-1031.zip](https://github.com/user-attachments/files/16321542/kmttg_20240720-1031.zip)

**Reply by tweiner** [reply]
Thank you so much.!!!!!
